pkg/secrets/secrets.go

```diff
@@ -87,7 +87,7 @@ func generateUserDataWithPubKey(pubKey string) string {
 			Start-Service sshd
 			$pubKeyConf = (Get-Content -path C:\ProgramData\ssh\sshd_config) -replace '#PubkeyAuthentication yes','PubkeyAuthentication yes'
 			$pubKeyConf | Set-Content -Path C:\ProgramData\ssh\sshd_config
- 			$passwordConf = (Get-Content -path C:\ProgramData\ssh\sshd_config) -replace '#PasswordAuthentication yes','PasswordAuthentication yes'
+ 			$passwordConf = (Get-Content -path C:\ProgramData\ssh\sshd_config) -replace '#PasswordAuthentication yes','PasswordAuthentication no'
```

Comment:
I realize that this will break if the default config ever changes i.e. `#PasswordAuthentication` is no longer present. But given it also applies to PubkeyAuthentication we can fix that in a follow up PR.